Makefile

```diff
@@ -50,6 +50,7 @@ PROXY_SERVER_IP ?= 127.0.0.1
 ## --------------------------------------
 mock_gen:
 	mkdir -p proto/agent/mocks
+	rm -f proto/agent/mocks/agent_mock.licensed.go proto/agent/agent_grpc.licensed.go
```

Comment:
I don't think we should need this line as a successful make should always move the file, hence cleaning it up. So I happy to have this in clean but I don't think it should be part of the normal make. The fact that we don't have client/proto/client/client_grpc.licensed.go here supports that.

If we do decide to keep it I would rather have this in the phony top level target can we move the deletes to the proper file targets? (Eg. `proto/agent/agent_grpc.pb.go proto/agent/agent.pb.go:`)

Reply:
You're right that it seems to not belong here.

I removed it and re-tested `make gen` a few times, at first it began mysteriously succeeding (where it failed before, with "unexpected EOF"). Then on the most recent try, it seemed to succeed but the resulting proto/agent/agent_grpc.pb.go was missing the license header. So, I think as-is there is some non-determinism.